File: python/models/batch_constants.py
"""Constants from BCHCON.cpy.

Mirrors the BATCH-CONTROL-CONSTANTS structure: status values, return-code
thresholds, process types, dependency types, and standard messages.
"""

from __future__ import annotations

from enum import Enum, IntEnum


class ProcessStatus(str, Enum):
    """BCT-STATUS values from BCHCON.cpy / BCHCTL.cpy."""

    READY = "R"
    ACTIVE = "A"
    WAITING = "W"
    DONE = "D"
    ERROR = "E"


class ReturnCode(IntEnum):
    """BCT-RC-* return-code thresholds from BCHCON.cpy."""

    SUCCESS = 0
    WARNING = 4
    ERROR = 8
    SEVERE = 12
    CRITICAL = 16


class ProcessType(str, Enum):
    """BCT-TYPE-* process types from BCHCON.cpy."""

    INITIAL = "INI"
    UPDATE = "UPD"
    REPORT = "RPT"
    CLEANUP = "CLN"


class DependencyType(str, Enum):
    """BCT-DEP-* dependency types from BCHCON.cpy."""

    REQUIRED = "R"
    OPTIONAL = "O"
    EXCLUSIVE = "X"


class RecordType(str, Enum):
    """BCT-REC-* control file record types from BCHCON.cpy."""

    CONTROL = "C"
    PROCESS = "P"
    DEPEND = "D"
    HISTORY = "H"


# Process control limits (BCT-CTRL-VALUES)
MAX_PREREQ = 10
MAX_RESTARTS = 3
WAIT_INTERVAL_SECONDS = 300
MAX_WAIT_SECONDS = 3600

# Standard messages (BCT-MESSAGES)
MSG_STARTING = "Process starting...           "
MSG_COMPLETE = "Process completed successfully"
MSG_FAILED = "Process failed - check errors "
MSG_WAITING = "Waiting for prerequisites     "

# Reserved process names (BCT-PROC-NAMES)
PROC_START_OF_DAY = "STARTDAY"
PROC_END_OF_DAY = "ENDDAY"
PROC_EMERGENCY = "EMERGENCY"

File: java/src/main/java/com/clbs/batch/BatchConstants.java
package com.clbs.batch;

/** BCHCON.cpy BATCH-CONTROL-CONSTANTS. */
public final class BatchConstants {

    public static final int RC_SUCCESS = 0;
    public static final int RC_WARNING = 4;
    public static final int RC_ERROR = 8;
    public static final int RC_SEVERE = 12;
    public static final int RC_CRITICAL = 16;

    public static final int MAX_PREREQ = 10;
    public static final int MAX_RESTARTS = 3;
    public static final int WAIT_INTERVAL = 300;
    public static final int MAX_WAIT_TIME = 3600;

    public static final String TYPE_INITIAL = "INI";
    public static final String TYPE_UPDATE = "UPD";
    public static final String TYPE_REPORT = "RPT";
    public static final String TYPE_CLEANUP = "CLN";

    public static final String DEP_REQUIRED = "R";
    public static final String DEP_OPTIONAL = "O";
    public static final String DEP_EXCLUSIVE = "X";

    public static final String START_OF_DAY = "STARTDAY";
    public static final String END_OF_DAY = "ENDDAY  ";
    public static final String EMERGENCY = "EMERGENCY";

    public static final String MSG_STARTING = "Process starting...           ";
    public static final String MSG_COMPLETE = "Process completed successfully";
    public static final String MSG_FAILED = "Process failed - check errors ";
    public static final String MSG_WAITING = "Waiting for prerequisites     ";

    private BatchConstants() {
    }
}
